recipes/aurelia-cli/build.sh

```diff
@@ -0,0 +1,14 @@
+#!/usr/bin/env bash
+
+set -o xtrace -o nounset -o pipefail -o errexit
+
```

Comment:
nodejs package with compiled code should have this boilerplate. Maybe we need to put this into a `{{ compiler('nodejs') }}` at one point. 

```suggestion
if [[ "${target_platform}" == "osx-arm64" ]]; then
  export npm_config_arch="arm64"
fi
# Don't use pre-built gyp packages
export npm_config_build_from_source=true

rm $PREFIX/bin/node
ln -s $BUILD_PREFIX/bin/node $PREFIX/bin/node
```

Reply:
Perhaps a `${NPM_ARGS}` variable would make sense for this case?  I can propose what should go in there in the `nodejs` feedstock for future submissions, though for now let's just set the boilerplate.

Also, according to here: https://github.com/swc-project/node-swc/issues/12, adding the flag `--build-from-source` to `npm install` is equivalent to setting `export npm_config_build_from_source=true`.  I have no strong preference for using a flag versus an environment variable, as long as we are consistent, but it's probably redundant to use both in these recipes.

I'll also note here that `--build-from-source` doesn't actually always guarantee that native **dependencies** will be built from source.  I carefully chose these packages because their native dependencies will build from source if the necessary build tools and other dependencies are available.  I currently have about 25 NodeJS packages where I haven't been to force rebuilding some native dependencies from source, though I'm steadily making progress.  I'm happy to share more info about that somewhere if it would be useful.

Reply:
> Perhaps a ${NPM_ARGS} variable would make sense for this case?

Yes, that sounds better. The `compiler('nodejs')` implementation would have the benefit that it could also do the symlinking for us.